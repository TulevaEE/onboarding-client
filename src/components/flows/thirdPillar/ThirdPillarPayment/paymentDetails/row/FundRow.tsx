import React from 'react';
import { FormattedMessage } from 'react-intl';

export const FundRow: React.FunctionComponent<{
  children: React.ReactNode;
}> = ({ children }) => (
  <tr>
    <td className="align-top text-right">{children}:</td>
    <td className="align-bottom pl-2">
      <b>
        <FormattedMessage id="thirdPillarPayment.tuleva3rdPillarFund" />
      </b>
    </td>
    <td className="d-none d-sm-table-cell" />
  </tr>
);
