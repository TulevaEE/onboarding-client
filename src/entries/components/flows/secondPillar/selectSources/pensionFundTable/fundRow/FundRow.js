import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { Message } from 'retranslate';

import Euro from '../../../../../common/Euro';

const FundRow = ({ price, name, highlighted, active }) => {
  const displayName = <Message>{name}</Message>;
  const displayPrice = <Euro amount={price} />;
  return (
    <div className="row tv-table__row py-2">
      <div className="col-12 col-sm">
        {highlighted ? <b>{displayName}</b> : displayName}
        {active ? '*' : ''}
      </div>
      <div className="col-12 col-sm text-sm-right">
        {highlighted ? <b>{displayPrice}</b> : displayPrice}
      </div>
      {/*
        <div className="col-12 col-sm text-sm-right">TODO</div>
         */}
    </div>
  );
};

FundRow.defaultProps = {
  price: 0,
  name: '',
  highlighted: false,
  active: false,
};

FundRow.propTypes = {
  price: Types.number,
  name: Types.oneOfType([Types.node, Types.string]),
  highlighted: Types.bool,
  active: Types.bool,
};

export default FundRow;
