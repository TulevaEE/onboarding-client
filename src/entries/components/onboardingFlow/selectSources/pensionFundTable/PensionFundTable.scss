@import "../../../variables";

.tv-table__header {
  font-size: 0.8em;
  background-color: $gray-washed;
  border-top-width: 1px;
  border-top-style: solid;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-color: $gray-mid;
}

.tv-table__row {
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-color: $gray-mid;
}
