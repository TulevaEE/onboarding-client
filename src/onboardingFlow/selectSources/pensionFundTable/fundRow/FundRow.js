import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { Message } from 'retranslate';

import { formatAmountForCurrency } from '../../../../common/utils';

const FundRow = ({ price, currency, name, highlighted, active }) => {
  const displayName = <Message>{name}</Message>;
  const displayPrice = formatAmountForCurrency(price, currency);
  return (
    <div className="row tv-table__row py-2">
      <div className="col-12 col-sm">
        { highlighted ? <b>{displayName}</b> : displayName }
        { active ? '*' : '' }
      </div>
      <div className="col-12 col-sm text-sm-right">
        { highlighted ? <b>{displayPrice}</b> : displayPrice }
      </div>
      {
        /*
        <div className="col-12 col-sm text-sm-right">TODO</div>
         */
      }
    </div>
  );
};

FundRow.defaultProps = {
  price: 0,
  currency: 'EUR',
  name: '',
  highlighted: false,
  active: false,
};

FundRow.propTypes = {
  price: Types.number,
  currency: Types.string,
  name: Types.oneOfType([Types.node, Types.string]),
  highlighted: Types.bool,
  active: Types.bool,
};

export default FundRow;
