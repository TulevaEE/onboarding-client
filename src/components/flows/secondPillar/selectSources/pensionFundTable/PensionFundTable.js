import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { FormattedMessage } from 'react-intl';

import FundRow from './fundRow';

import './PensionFundTable.scss';
import { getTotalFundValue } from '../../../../common/utils';

const PensionFundTable = ({ funds }) => {
  const totalPrice = getTotalFundValue(funds);
  return (
    <div>
      <div className="row tv-table__header py-2">
        <div className="col-12 col-sm">
          <FormattedMessage id="select.sources.pension.fund" />
        </div>
        <div className="col-12 col-sm text-sm-right">
          <FormattedMessage id="select.sources.value" />
        </div>
      </div>
      {funds &&
        funds.length > 0 &&
        funds.map(({ currency, price, name, isin, activeFund }) => (
          <FundRow key={isin} price={price} name={name} currency={currency} active={activeFund} />
        ))}
      <FundRow
        price={totalPrice}
        currency="EUR" // hardcoded until there are more currencies
        name="select.sources.total"
        highlighted
      />
      <div className="mt-2">
        <small className="text-muted">
          <FormattedMessage id="select.sources.active.fund" />
        </small>
      </div>
    </div>
  );
};

PensionFundTable.defaultProps = {
  funds: [],
};

PensionFundTable.propTypes = {
  funds: Types.arrayOf(
    Types.shape({
      price: Types.number,
      currency: Types.string,
      name: Types.string,
      manager: Types.string,
      isin: Types.string,
      activeFund: Types.bool,
    }),
  ),
};

export default PensionFundTable;
