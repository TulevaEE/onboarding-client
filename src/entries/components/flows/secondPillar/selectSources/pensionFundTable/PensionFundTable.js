import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { Message } from 'retranslate';

import FundRow from './fundRow';

import './PensionFundTable.scss';
import { getTotalFundValue } from '../../../../common/utils';

const PensionFundTable = ({ funds }) => {
  const totalPrice = getTotalFundValue(funds);
  return (
    <div>
      <div className="row tv-table__header py-2">
        <div className="col-12 col-sm">
          <Message>select.sources.pension.fund</Message>
        </div>
        <div className="col-12 col-sm text-sm-right">
          <Message>select.sources.value</Message>
        </div>
        {/*
          <div className="col-12 col-sm text-sm-right">
            <Message>select.sources.fees</Message>
          </div>
           */}
      </div>
      {funds &&
        funds.length &&
        funds.map(({ currency, price, name, isin, activeFund }) => (
          <FundRow key={isin} price={price} name={name} currency={currency} active={activeFund} />
        ))}
      <FundRow
        price={totalPrice}
        currency="EUR" // hardcoded until there are more currencies
        name="select.sources.total"
        highlighted
      />
      <div className="mt-2">
        <small className="text-muted">
          <Message>select.sources.active.fund</Message>
        </small>
      </div>
    </div>
  );
};

PensionFundTable.defaultProps = {
  funds: [],
};

PensionFundTable.propTypes = {
  funds: Types.arrayOf(
    Types.shape({
      price: Types.number,
      currency: Types.string,
      name: Types.string,
      manager: Types.string,
      isin: Types.string,
      activeFund: Types.bool,
    }),
  ),
};

export default PensionFundTable;
