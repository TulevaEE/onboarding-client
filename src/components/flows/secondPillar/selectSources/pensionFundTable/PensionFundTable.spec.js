import React from 'react';
import { shallow } from 'enzyme';
import { FormattedMessage } from 'react-intl';

import PensionFundTable from './PensionFundTable';
import FundRow from './fundRow';

describe('Pension fund table', () => {
  let component;

  beforeEach(() => {
    component = shallow(<PensionFundTable />);
  });

  it('renders a table header', () => {
    const headerFields = () => component.find('div.tv-table__header').children();
    expect(headerFields()).toMatchSnapshot();
  });

  it('renders a fund row for every fund', () => {
    const funds = [
      { name: 'fund 1', isin: 'isin 1', price: 1, currency: 'EUR', pillar: 2 },
      { name: 'fund 2', isin: 'isin 2', price: 2, currency: 'EUR', pillar: 2 },
    ];
    component.setProps({ funds });
    expect(component.find(FundRow).at(0).get(0)).toEqual(
      <FundRow name="fund 1" key="isin 1" price={1} currency="EUR" />,
    );
    expect(component.find(FundRow).at(1).get(0)).toEqual(
      <FundRow name="fund 2" key="isin 2" price={2} currency="EUR" />,
    );
  });

  it('renders a tiny legend under the table', () => {
    expect(component.contains(<FormattedMessage id="select.sources.active.fund" />)).toBe(true);
  });

  it('renders a fundrow for totals', () => {
    const funds = [
      { name: 'fund 1', isin: 'isin 1', price: 1, currency: 'EUR', pillar: 2 },
      { name: 'fund 2', isin: 'isin 2', price: 2, currency: 'EUR', pillar: 2 },
    ];
    component.setProps({ funds });
    expect(component.find(FundRow).at(2).get(0)).toEqual(
      <FundRow name="select.sources.total" price={3} currency="EUR" highlighted />,
    );
  });
});
