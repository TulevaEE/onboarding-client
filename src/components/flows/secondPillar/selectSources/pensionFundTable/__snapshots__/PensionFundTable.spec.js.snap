// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pension fund table renders a table header 1`] = `
Array [
  <div
    className="col-12 col-sm"
  >
    <MemoizedFormattedMessage
      id="select.sources.pension.fund"
    />
  </div>,
  <div
    className="col-12 col-sm text-sm-right"
  >
    <MemoizedFormattedMessage
      id="select.sources.value"
    />
  </div>,
]
`;
